src/browser/KommunicateCordovaPluginProxy.js

```diff
@@ -0,0 +1,302 @@
+				}
+			}
+			initPlugin(kmUser, (response) => {
+        if(!(kmUser.withPreChat && kmUser.withPreChat == true)) {
+         createConversation(conversationObj, kmUser.userId, successCallback, errorCallback);
+        }
+			}, (error) => {
+				errorCallback(error);
+			});
+		}
+	},
+
+	logout: function (successCallback, errorCallback) {
+		console.log("Called logout function ");
+		if (isUserLoggedIn() && typeof Kommunicate != 'undefined' && Kommunicate) {
+			init((response) => {
+				//Kommunicate.launchConversation();
+				Kommunicate.logout();
+				localStorage.removeItem('KM_PLUGIN_USER_DETAILS');
+				location.reload()
+				successCallback(response)
+			}, (error) => {
+				errorCallback(error)
+			});
+		} else {
+			localStorage.removeItem('KM_PLUGIN_USER_DETAILS');
+			successCallback("success")
+			location.reload()
+		}
+	},
+
+	processPushNotification: function (successCallback, errorCallback, data) {
+		console.log("Called processPushNotification function : " + data);
+		errorCallback("function not implemented");
+	},
+
+	updatePushNotificationToken: function (successCallback, errorCallback, token) {
+		console.log("Called updatePushNotificationToken function : " + token);
+		errorCallback("function not implemented");
+	},
+
+	registerPushNotification: function (successCallback, errorCallback) {
+		console.log("Called registerPushNotification function ");
+		errorCallback("function not implemented");
+	}
+}
+
+function init(successCallback, errorCallback) {
+	if (isUserLoggedIn()) {
+		if (typeof Kommunicate != 'undefined' && Kommunicate) {
+			successCallback("success")
+		} else {
+			initPlugin(null, successCallback, errorCallback)
+		}
+	} else {
+		errorCallback("User not logged in, call login first")
+	}
+}
+
+function initPlugin(kmUser, successCallback, errorCallback) {
+	if (localStorage && localStorage.KM_PLUGIN_USER_DETAILS) {
+		kmUser = JSON.parse(localStorage.KM_PLUGIN_USER_DETAILS)
+	}
+
+	(function (d, m) {
+		var kommunicateSettings = {
+			"appId": kmUser.applicationId,
+			"popupWidget": false,
+			"automaticChatOpenOnNavigation": false,
+			"userId": kmUser.userId,
+			"password": kmUser.password,
+			"userName": kmUser.displayName,
+			"email": kmUser.email,
+			"imageLink": kmUser.imageLink,
+			"preLeadCollection": kmUser.withPreChat ? getPrechatLeadDetails() : [],
+			"authenticationTypeId": kmUser.authenticationTypeId,
+			"onInit": function (response) {
+				if (response && response === "success") {
+					if (kmUser.withPreChat == true) {
+						kmUser.userId = JSON.parse(sessionStorage.getItem("mckAppHeaders")).userId;
+          }
+          
+          KommunicateGlobal.document.getElementById('km-chat-widget-close-button').addEventListener('click',function(){
+            var testClick = parent.document.getElementById("kommunicate-widget-iframe");
+            testClick.style.display = "none";
+          });
+
+					console.log("Login response : " + JSON.stringify(response));
```

Comment:
@reytum can this be disabled, some customers have raised concern about printing user's info in console

Reply:
Will remove the logs as there is no option to restrict logs for productions and localhost in web.